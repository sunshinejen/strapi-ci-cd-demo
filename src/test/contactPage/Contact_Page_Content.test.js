import React from 'react'
import { render, screen } from '@testing-library/react';
import ContactPage  from '../../components/contact/contactcontent'

//Testing Data
const contactPageData = {title: 'Contact Us', content: "Harvard In tech Contact page is working"}

// Smoke test
it('Contace pake links renders', () => {
    render(<ContactPage contactPageData={contactPageData}/>)
});

//Snapshot Test
test("component matches snapshot", () => {
    const { asFragment } = render(<ContactPage contactPageData={contactPageData} />)
    expect(asFragment()).toMatchSnapshot()
});

// Unit test
test('Contact page Links unit test', () => {
    const {getByText} = render(<ContactPage contactPageData={contactPageData}/>)

    expect(getByText('Contact Us')).toBeInTheDocument
    expect(getByText("Harvard In tech Contact page is working")).toBeInTheDocument

    expect(screen.getByTestId('linkedin-link').href).toBe("https://www.linkedin.com/company/harvard-in-tech-seattle/")
    expect(screen.getByTestId('twitter-link').href).toBe("https://twitter.com/in_harvard")
    expect(screen.getByTestId('facebook-link').href).toBe("https://www.facebook.com/profile.php?id=100070986468967")
    expect(screen.getByTestId('email-link').href).toBe("mailto:info@harvardintechseattle.com")
    expect(screen.getByTestId('volunteer-link').href).toBe("https://lnkd.in/g43ipkP")
});