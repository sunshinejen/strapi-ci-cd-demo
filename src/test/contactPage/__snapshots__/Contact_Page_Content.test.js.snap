// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="illustration-header d-flex align-items-center mb-5"
    >
      <div
        class="content"
      >
        <h4
          class="contact-page-title"
        >
          Contact Us
        </h4>
        <p
          class="contact-page-content"
        >
          Harvard In tech Contact page is working
        </p>
        <div
          class="d-flex flex-wrap"
        >
          <div
            class="text-center"
          >
            <a
              data-testid="linkedin-link"
              href="https://www.linkedin.com/company/harvard-in-tech-seattle/"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              <span
                class="fa-stack fa-2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle fa-w-16  fa-stack-2x"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                    fill="currentColor"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-linkedin fa-w-14 fa-stack-1x fa-inverse"
                  data-icon="linkedin"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p>
                LinkedIn
              </p>
            </a>
          </div>
          <div
            class="text-center"
          >
            <a
              data-testid="twitter-link"
              href="https://twitter.com/in_harvard"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              <span
                class="fa-stack fa-2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle fa-w-16  fa-stack-2x"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                    fill="currentColor"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-twitter fa-w-16 fa-inverse  fa-stack-1x fa-inverse"
                  data-icon="twitter"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p>
                Twitter
              </p>
            </a>
          </div>
          <div
            class="text-center"
          >
            <a
              data-testid="facebook-link"
              href="https://www.facebook.com/profile.php?id=100070986468967"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              <span
                class="fa-stack fa-2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle fa-w-16  fa-stack-2x"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                    fill="currentColor"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-facebook-f fa-w-10 fa-inverse  fa-stack-1x fa-inverse"
                  data-icon="facebook-f"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p>
                Facebook
              </p>
            </a>
          </div>
          <div
            class="text-center"
          >
            <a
              data-testid="email-link"
              href="mailto:info@harvardintechseattle.com"
              style="text-decoration: none;"
            >
              <span
                class="fa-stack fa-2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle fa-w-16  fa-stack-2x"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                    fill="currentColor"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope fa-w-16 fa-inverse  fa-stack-1x fa-inverse"
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p>
                Email
              </p>
            </a>
          </div>
          <div
            class="text-center"
          >
            <a
              data-testid="volunteer-link"
              href="https://lnkd.in/g43ipkP"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              <span
                class="fa-stack fa-2x"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-circle fa-w-16  fa-stack-2x"
                  data-icon="circle"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                    fill="currentColor"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-user fa-w-14 fa-inverse fa-stack-1x fa-inverse"
                  data-icon="user"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              <p>
                Volunteer
              </p>
            </a>
          </div>
        </div>
      </div>
      <img
        alt="Contact Clipart"
        class="img-fluid"
        src="test-file-stub"
      />
    </div>
  </div>
</DocumentFragment>
`;
